capitalone/rubicon-ml
# config/dotfile consolidation

## What
  * consolidating dotfiles and config files into pyproject.toml
    * move .coveragerc into pyproject.toml
    * move setup.py into pyproject.toml
    * move most of setup.cfg into pyproject.toml
      * flake8 doesn't support toml configuration - I propose we switch to pylint in a followup
  * consolidating environment yamls into a single requirements.txt
    * optional dependencies now configured entirely in pyproject.toml
  * update workflows to reflect changes
  * update docs/readmes to reflect changes

## How to Test
  * create a new environment locally and `pip install -e ".[dev]"` then run...
    * `python -m pytest` to validate the installation and the tests pass
    * `pre-commit run --all-files` to validate the dev tool configuration in pyproject.toml
  * validate the workflows all run
    * test-package: https://github.com/capitalone/rubicon-ml/actions/runs/6226019174?pr=382
      * the "Validate build" stage should validate the publish-package action
    * edgetest: https://github.com/capitalone/rubicon-ml/actions/runs/6226023458/job/16897640509
    * publish-docs: https://github.com/capitalone/rubicon-ml/actions/runs/6227657457/job/16902751587
  * create another new environment locally and `pip install -e ".[docs]"` then run...
    * `cd docs && make clean html open`
